### PR TITLE
Add audiotest and videotest driver

### PR DESCRIPTION
**Pull request by at-wat**
Which work similar to GStreamer audiotestsrc, videotestsrc.

image

### CODE REVIEW

File: pkg/driver/videotest/dummy.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		{107, 202, 222},
+		{82, 90, 240},
+		{41, 240, 110},
+	}
+
+	r := video.ReaderFunc(func() (image.Image, error) {
+		select {
+		case <-d.closed:
+			return nil, io.EOF
+		default:
```
Comment (at-wat) [inline pkg/driver/videotest/dummy.go]: need to add ticker

**Comment by at-wat** [comment]
It may be nice to add an instruction to use this in the examples, for users who don't have camera right now.

**Comment by at-wat** [comment]
rebased

File: pkg/driver/videotest/dummy.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	"github.com/pion/mediadevices/pkg/driver"
+	"github.com/pion/mediadevices/pkg/frame"
+	"github.com/pion/mediadevices/pkg/io/video"
+	"github.com/pion/mediadevices/pkg/prop"
+)
+
+func init() {
+	driver.GetManager().Register(
+		newVideoTest(),
+		driver.Info{Label: "VideoTest", DeviceType: driver.Microphone},
```
Comment (lherman-cs) [inline pkg/driver/videotest/dummy.go]: Oops, it should be `driver.Camera` there

File: pkg/driver/audiotest/dummy.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+func init() {
+	driver.GetManager().Register(
+		&dummy{}, driver.Info{Label: "AudioTest", DeviceType: driver.Microphone},
+	)
+}
+
+type dummy struct {
+	closed <-chan struct{}
+	cancel func()
+	phase  int
```
Comment (lherman-cs) [inline pkg/driver/audiotest/dummy.go]: phase can be just inside AudioRecord, less scope.

File: pkg/driver/videotest/dummy.go (hunk 2 above)
Comment (at-wat) [reply pkg/driver/videotest/dummy.go]: oops

**Comment by lherman-cs** [comment]
> It may be nice to add an instruction to use this in the examples, for users who don't have camera right now.

That's a good point. How about if we just add a comment in the simple example, above the camera and microphone imports?

Something like:
```
	// Note: If you don't have a camera or microphone or your adapters are not supported,
	//       you can always swap your adapters with our dummy adapters below.
	// _ "github.com/pion/mediadevices/pkg/driver/videotest"
	//_ "github.com/pion/mediadevices/pkg/driver/audiotest"
	_ "github.com/pion/mediadevices/pkg/driver/camera"     // This is required to register camera adapter
	_ "github.com/pion/mediadevices/pkg/driver/microphone" // This is required to register microphone adapte
```

**Comment by at-wat** [comment]
https://github.com/pion/mediadevices/pull/48#issuecomment-586555954 sounds very nice:+1:

**Comment by at-wat** [comment]
Addressed you comments! I'll rebase and add the comment to the simple example.

**Pull request merged**